### PR TITLE
Change the behaviour of the subject attribute dropdown for SAML IdPs

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose
- This PR will disable the subject attribute dropdown in SAML IdPs when there are no attribute mappings are present. It will consider the default subject attribute when there are no mappings.
Screenshot 2021-09-21 at 13 43 10

- The dropdown will be enabled when the attribute mappings are present.
Screenshot 2021-09-21 at 13 45 07

- This PR also fixes the issue where attribute settings are not properly updated at times by making 2 parallel API calls to occur sequentially.

### Related Issues
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx [hunk 1]
```diff
@@ -78,9 +83,16 @@ export const UriAttributesSettings: FunctionComponent<AdvanceAttributeSettingsPr
...
                     <Form>
                         <Form.Select
-                            required
                             fluid
-                            options={ dropDownOptions }
+                            options={ 
+                                dropDownOptions.concat(
+                                    {
+                                        key: "default_subject",
+                                        text: "Default Subject",
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx:91]: Let's add i18n placeholder

File: apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx [hunk 2]
```diff
@@ -102,10 +114,14 @@ export const UriAttributesSettings: FunctionComponent<AdvanceAttributeSettingsPr
                                 pointing: "above"
                             } }
                             readOnly={ isReadOnly }
+                            disabled={ isMappingEmpty }
                         />
                     </Form>
                     <Hint>
-                        { t("console:develop.features.authenticationProvider.forms.uriAttributeSettings.subject.hint") }
+                        The attribute that identifies the user at the enterprise identity provider.
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx:121]: Let's add i18n.

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx (hunk 1 above)
Comment (DonOmalVindula) [reply apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx]: Fixed with fb82878ef19d02b18fe3f6acd2e9908a1a336e3a

File: apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx (hunk 2 above)
Comment (DonOmalVindula) [reply apps/console/src/features/identity-providers/components/settings/attribute-management/uri-attributes-settings.tsx]: Fixed with fb82878ef19d02b18fe3f6acd2e9908a1a336e3a

**Pull request merged**